### ISSUE TITLE
mediKanren export of KG2.3.5

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#1121]
Now that we have a build of KG2.3.5, this might be a good juncture to try exporting it to CSV and mediKanren `.scm` files following the instructions in `RTX/code/kg2/mediKanren`. We can then copy the CSV and `.scm` files to `s3://rtx-kg2-public` to make them available to Team Unsecret Agent. Since Unsecret Agent already has our build of KG2.3.4, this update is probably not urgent; more of a when-there-is-time kind of thing. I'm also interested in getting an independent test of the export process and instructions (any issues or points where the documentation needs clarification can be reported in the GitHub issue tracker and tagged `kg2`).

**Comment by kvarforl** [comment]
Listing other issues to be considered while doing this: #1073, #1079, #1056

**Comment by kvarforl** [comment]
I figured I would investigate #1073 and see if it saves us time to use mediKanren's `tsv-graph-to-db.rkt` script (instructions [here](https://github.com/webyrd/mediKanren/tree/master/biolink#converting-tsvs-or-csvs-to-medikanren-format)) instead of going from tsv to csv and then creating the indexes.

However, [our fork](https://github.com/RTXteam/mediKanren) of the mediKanren repo is really behind the original, so it doesn't have that script yet... is there any reason I shouldn't update it to be in sync? @finnagin

**Comment by kvarforl** [comment]
Okay, so I tried using the latest version of the mediKanren repository to create the indexes and verify them, but for some reason I think the `create-index.sh` script didn't complete. 
I figured this out because I'm missing at least one  .scm file. Following the [Testing the Indexes](https://github.com/RTXteam/RTX/tree/master/code/kg2/mediKanren#testing-the-indexesl) instructions results in this error: 
```
Welcome to Racket v6.11.
> (require "mk-db.rkt")
> (define rtx2 (make-db "data/rtx_kg2"))
; open-input-file: cannot open input file
;   path:
;     /home/ubuntu/kg2-code/mediKanren/mediKanren/biolink/data/rtx_kg2/concepts-by-synonym.scm
;   system error: No such file or directory; errno=2
; [,bt for context] 
```
and the log file for  `create-index.sh` doesn't contain `+racket build-string-index.rkt data rtx_kg2`, which makes me think it died during `csv-graph-to-db.rkt`. (the last line of the log file says `Processed 32000000 edges`, but KG2.3.5 has  53,456,505 edges)

So long story short... I think I'm just going to try again and see what happens.

**Comment by kvarforl** [comment]
FYI, I made it so I can flip between the newest up to date mediKanren repo and our older one by adding the original repo as an upstream source, but I haven't merged the two yet.
`git checkout master` gets me to the out of date version
`git checkout upstream/master` gets me to the newest version

**Comment by kvarforl** [comment]
Updates on this: 
When running `create index.sh` on the .csv files generated from `kg2_tsv_to_medikanren_csv.py`,  the `csv-graph-to-db.rkt` consistently silently fails after processing around 32 million edges (specifically, `edges.scm` has 32088675 edges when it fails. it dies around 3.5 hours after starting) 

The `kg2_tsv_to_medikanren_csv.py` completes successfully because `rtx_kg2.edge.csv` has 53456505 edges, which aligns with  the notes from the build. I've regenerated the csv's using this method, and the same failure in the racket script occurs.

I also attempted to use to KGX repo to generate the CSV files (once using our out of date specific commit, and once using the most recent commit on the upstream's source-sink branch) but both ran out of memory before completing.

I think my next move is to investigate that specific edge that it seems to fail on.

Tabling this for a moment while I work on getting KG2.3.4 up, but will revisit :)

**Comment by saramsey** [comment]
Hey @finnagin, @kvarforl  has found an interesting issue with `create-index.sh` in the `mediKanren` code; can you take a look at this issue and coordinate with @kvarforl ?

**Comment by finnagin** [comment]
@saramsey Sure! I can take a look. Which ec2 instance and what directory are you running `create index.sh` in @kvarforl ?

**Comment by kvarforl** [comment]
@finnagin I think this is the same issue I had you look into a few weeks ago! I tried to run it on kg2lindsey.rtx.ai, from the `RTX/code/kg2/mediKanren` directory, if you'd like to take a look again. 

From [this](https://github.com/RTXteam/RTX/issues/1118#issuecomment-738304183) comment I think the consensus with @saramsey is that it's probably not vital to get a mediKanren export of KG2.3.5, but I imagine the same issue might persist with KG2.4.0 once it's built :)

**Comment by saramsey** [comment]
Hi @finnagin and @kvarforl where are we with this issue?  Can it be closed out?

**Comment by kvarforl** [comment]
Yes! I left it open for reference when I do the KG2.5.1 export, which I'm working on now. No reason I can't just look at the closed issue haha :)
